ts/index.ts

```diff
@@ -146,7 +152,7 @@ class Gossipsub extends Pubsub {
    */
   constructor (
     libp2p: Libp2p,
-    options: Partial<GossipInputOptions> = {}
+    options: Partial<GossipInputOptions> & Pick<GossipInputOptions, 'fastMsgIdFn'>
```

Comment:
We should allow undefined `fastMsgIdFn`

And default to `(msg) => messageIdToString(options.msgIdFn(msg))`

Reply:
@wemeetagain `options. msgIdFn` is also optional and I don't see it being used, is that an issue? I suppose we should:
+ remove `msgIdFn` in `GossipInputOptions`
+ default the fast msg id fn to `(msg) => messageIdToString(await this.getMsgId(msg))`